Data/ByteString.hs

```diff
@@ -1213,7 +1213,22 @@ groupBy k xs = case uncons xs of
 -- 'ByteString's and concatenates the list after interspersing the first
 -- argument between each element of the list.
 intercalate :: ByteString -> [ByteString] -> ByteString
-intercalate s = concat . List.intersperse s
+intercalate _ [] = mempty
+intercalate (BS f_sep_ptr sep_len) (BS f_h_ptr h_len : t) =
+  unsafeCreate total_len $ \dst_ptr0 ->
+    unsafeWithForeignPtr f_sep_ptr $ \sep_ptr -> do
+      unsafeWithForeignPtr f_h_ptr $ \h_ptr ->
+        memcpy dst_ptr0 h_ptr (fromIntegral h_len)
```

Comment:
Please adorn `fromIntegral` woth type applications, explaining what kind of conversion is going on, e. g., `fromIntegral @Int @CInt`.

Reply:
Looks like quite a few of those were redundant in that file, I've removed them.